src/ec/dasharo/ec/acpi/dshr.asl

```diff
@@ -0,0 +1,147 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include "../driver.h"
+
+Device (DASH) {
+	Name (_HID, "DSHR0001")
+	Name (_UID, 0)
+	// Hide the device so that Windows does not warn about a missing driver.
+	Name (_STA, 0xB)
+
+	// Version of the ACPI interface
+	Name (DSVR, 1)
+
+	// Get platform features
+	Method (GFTR, 0, Serialized)
+	{
+		Return (
+			(1 << DASHARO_FEATURE_TEMPERATURE) |
+			(1 << DASHARO_FEATURE_FAN_PWM) |
+			(1 << DASHARO_FEATURE_FAN_TACH))
+	}
+
+	// Get feature capabilities
+	Method (GFCP, 2, Serialized)
+	{
+		Switch (ToInteger(Arg0))
+		{
+			Case (DASHARO_FEATURE_TEMPERATURE)
+			{
+				Switch (ToInteger(Arg1))
+				{
+					Case (DASHARO_TEMPERATURE_CPU_PACKAGE) { Return (1) }
+#if CONFIG(EC_DASHARO_EC_DGPU)
+					Case (DASHARO_TEMPERATURE_GPU) { Return (1) }
+#endif
+					Default { Return (0) }
+				}
+			}
+
+			Case (DASHARO_FEATURE_FAN_PWM)
+			{
+				Switch (ToInteger(Arg1))
+				{
+					Case (DASHARO_FAN_CPU) { Return (CONFIG_EC_DASHARO_EC_CPU_FAN_COUNT) }
```

Comment:
Wouldn't this mean that if EC_DASHARO_EC_DGPU is enabled, then EC_DASHARO_EC_CPU_FAN_COUNT must be 1? Currently it is possible to misconfigure, e.g. EC_DASHARO_EC_DGPU=1 and EC_DASHARO_EC_CPU_FAN_COUNT=2

Reply:
Done in https://github.com/Dasharo/coreboot/pull/646/commits/d7ffd78a91e4c2a3dc93eb78e238b2c381723c81